cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/LogAppender.java

```diff
@@ -69,8 +70,8 @@ public final void append(ILoggingEvent eventObject) {
       if (loggingContext == null) {
         return;
       }
+      eventObject = addExtraTags(eventObject, loggingContext);
```

Comment:
should be adding all the tags in one method instead of relying on the fact that this call makes the event modifiable (the tags should also not be modifiable).

Reply:
I mean what was commented in the DelegatingLoggingEvent about wrapping the tags in an unmodifiable map.